framework/src/main/java/org/checkerframework/common/basetype/BaseTypeVisitor.java

```diff
@@ -1584,10 +1602,138 @@ && getCurrentPath().getParentPath().getLeaf().getKind()
             // so only validate if commonAssignmentCheck wasn't called
             validateTypeOf(tree);
         }
+        validateVariablesTargetLocation(tree, variableType);
         warnRedundantAnnotations(tree, variableType);
         return super.visitVariable(tree, p);
     }
 
+    /**
+     * Validate if the annotations on the VariableTree are at the right locations, which is
+     * specified by the meta-annotation @TargetLocations. The difference of this method between
+     * {@link BaseTypeVisitor#validateTargetLocation(Tree, AnnotatedTypeMirror, TypeUseLocation)} is
+     * that this one is only used in {@link BaseTypeVisitor#visitVariable(VariableTree, Void)}
```

Comment:
But why? What are the conceptual differences between this method and the other one? Please make a separate PR that improves on this.

Reply:
opened a new PR eisop#582 for this. Continue the discussion there.